File: IntersectionOfTwoArrays.py
def intersection(a, b):
    # Chuyển mảng a thành set (tập hợp) để loại bỏ phần tử trùng lặp
    sa = set(a)

    res = []  # Danh sách lưu kết quả giao của 2 mảng

    # Duyệt mảng b
    for elem in b:
        # Nếu phần tử trong b có mặt trong set sa
        if elem in sa:
            res.append(elem)  # Thêm phần tử vào kết quả
            sa.remove(elem)   # Xóa phần tử khỏi set để tránh thêm lại phần tử trùng

    return res  # Trả về kết quả

# Nhập mảng từ người dùng
a = list(map(int, input("Nhập mảng a (các phần tử cách nhau bởi dấu cách): ").split()))
b = list(map(int, input("Nhập mảng b (các phần tử cách nhau bởi dấu cách): ").split()))

# Gọi hàm intersection để tìm phần tử chung
res = intersection(a, b)

# In kết quả
print("Phần tử chung giữa 2 mảng:", " ".join(map(str, res)))

File: IntersectionOfTwoArrays.c
#include <stdio.h>
#include <stdlib.h>
// Hàm so sánh để sắp xếp mảng
int cmp(const void *a, const void *b) {
    return (*(int *)a - *(int *)b);
}
// Hàm tìm phần tử chung của 2 mảng
void intersection(int a[], int b[], int n, int m) {
    // Sắp xếp mảng a và mảng b
    qsort(a, n, sizeof(int), cmp);
    qsort(b, m, sizeof(int), cmp);

    // Duyệt mảng a và b để tìm phần tử chung
    int i = 0, j = 0;
    printf("Phần tử chung của 2 mảng là: ");
    while (i < n && j < m) {
        if (a[i] == b[j]) {
            printf("%d ", a[i]);
            i++;
            j++;
        } else if (a[i] < b[j]) {
            i++;
        } else {
            j++;
        }
    }
    printf("\n");
}

int main() {
    int n, m;

    // Nhập kích thước và phần tử của mảng a
    printf("Nhập kích thước mảng a: ");
    scanf("%d", &n);
    int a[n];
    printf("Nhập phần tử mảng a: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }

    // Nhập kích thước và phần tử của mảng b
    printf("Nhập kích thước mảng b: ");
    scanf("%d", &m);
    int b[m];
    printf("Nhập phần tử mảng b: ");
    for (int i = 0; i < m; i++) {
        scanf("%d", &b[i]);
    }

    // Gọi hàm intersection để tìm phần tử chung
    intersection(a, b, n, m);

    return 0;
}
